book/website/communication/binder/zero-to-binder.md

```diff
@@ -0,0 +1,525 @@
+
+```python
+%matplotlib inline
+
+import pandas
+
+data = pandas.read_csv("gapminder.csv", index_col="country")
+
+years = data.columns.str.strip("gdpPercap_")  # Extract year from last 4 characters of each column name
+data.columns = years.astype(int)              # Convert year values to integers, saving results back to dataframe
+
+data.loc["Australia"].plot()
+```
+
+```{note}
+See this [Software Carpentry lesson](https://swcarpentry.github.io/python-novice-gapminder/09-plotting/index.html) for more info.
+```
+````
+
+````{tabbed} Julia
+1) Go to your GitHub repo and create a file called `postBuild`
+2) In `postBuild`, add a single line reading: `wget -q -O gapminder.csv http://bit.ly/2uh4s3g`
+   - `wget` is a program which retrieves content from web servers.
+     This line extracts the content from the bitly URL and saves it to the filename denoted by the `-O` flag (capital "O", not zero), in this case `gapminder.csv`.
+     The `-q` flag tells `wget` to do this quietly, meaning it won't print anything to the console.
+3) Update your `Project.toml` file by adding new dependencies to `[deps]` with the following lines:
+
+   ```julia
+   DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
+   Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
+   ```
+
+   - These packages aren't necessary to download the data but we will use them to read the CSV file and make a plot
+4) Click the binder badge in your README to launch your Binder
+
+Once the Binder has launched, you should see a new file has appeared that was not part of your repo when you clicked the badge.
+
+Now visualise the data by creating a new notebook (selecting "Julia" from the Notebook section) and run the following code in a cell.
+
+```julia
+using DataFrames
+using CSV
+using Plots
+
+data = CSV.read("gapminder.csv", DataFrame)
+
+# Extract the row corresponding to Australia
+aus_gdp = data[data[:, :country] .== "Australia", :]
+aus_gdp = Matrix(aus_gdp[:,2:end])[:]  # as vector
+
+# Extract the years as Ints from the column names
+years = [x[end-3:end] for x in names(data)[2:end]]
+years = parse.(Int, years)
+
+# Plot
+plot(years, aus_gdp)
+```
+````
+
+````{tabbed} R
+1) Go to your GitHub repo and create a file called `postBuild`
+2) In `postBuild`, add a single line reading: `wget -q -O gapminder.csv http://bit.ly/2uh4s3g`
+   - `wget` is a program which retrieves content from web servers.
+     This line extracts the content from the bitly URL and saves it to the filename denoted by the `-O` flag (capital "O", not zero), in this case `gapminder.csv`.
+     The `-q` flag tells `wget` to do this quietly, meaning it won't print anything to the console.
+3) Update your `install.R` file to install two additional dependencies, `"tidyr"` and `"ggplot2"`. To do so, supply a character vector of the required packages to `install.packages()` instead of a single character string. The installation command should now look like this:
```

Comment:
I would also add `dplyr` to the mix and make the code below more `tidyverse`-like using the `filter` function.

Reply:
So my vision for the multi-language workshop, is that there should be as few differences as possible between the tracks except what you feed to binder. I'm also not teaching them anything about coding best practices (or indeed, git best practices! All the commits to main!). So I kinda don't really know what side of the fence I come down on on this point.